okhttp/src/main/java/okhttp3/internal/platform/Platform.kt

```diff
@@ -80,27 +77,17 @@ open class Platform {
   /** Prefix used on custom headers.  */
   fun getPrefix() = "OkHttp"
 
-  open fun getSSLContext(): SSLContext = try {
-    SSLContext.getInstance("TLS")
-  } catch (e: NoSuchAlgorithmException) {
-    throw IllegalStateException("No TLS provider", e)
-  }
+  open fun newSSLContext(): SSLContext = SSLContext.getInstance("TLS")
 
   open fun platformTrustManager(): X509TrustManager {
-    try {
-      val trustManagerFactory = TrustManagerFactory.getInstance(
-          TrustManagerFactory.getDefaultAlgorithm())
-      trustManagerFactory.init(null as KeyStore?)
-      val trustManagers = trustManagerFactory.trustManagers
-      if (trustManagers.size != 1 || trustManagers[0] !is X509TrustManager) {
-        throw IllegalStateException(
-            "Unexpected default trust managers:" + Arrays.toString(trustManagers))
-      }
-      return trustManagers[0] as X509TrustManager
-    } catch (e: GeneralSecurityException) {
-      throw AssertionError("No System TLS", e) // The system has no TLS. Just give up.
+    val factory = TrustManagerFactory.getInstance(
+        TrustManagerFactory.getDefaultAlgorithm())
+    factory.init(null as KeyStore?)
+    val trustManagers = factory.trustManagers!!
+    check(trustManagers.size == 1 && trustManagers[0] is X509TrustManager) {
```

Comment:
TIOLI: If writing in kotlin from scratch I might go for

```
      val tmf = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm()).apply { 
        init(null as KeyStore?)
      }
      
      return tmf.trustManagers?.firstOrNull() as? X509TrustManager ?: throw IllegalStateException("XXX")
```

Mainly to focus on the good path, not the bad